Homework2/Trie/Trie.cs

```diff
@@ -0,0 +1,198 @@
+﻿namespace Trie;
+
+using System;
+
+// Trie class that implements Trie data structure.
+internal class Trie
+{
+    TrieNode root;
```

Comment:
```suggestion
    private TrieNode root;
```
Модификаторы доступа пишутся всегда, даже если умолчания устраивают. Тут и ниже. Это напрямую связано с тем, что в разных языках умолчания разные (например, в Java по умолчанию пакетная видимость).